build_cmake/Framework/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+cmake_minimum_required(VERSION 2.8)
+project(Framework)
+
+set(CMAKE_MODULE_PATH
+	${CMAKE_CURRENT_SOURCE_DIR}/../../../Dependencies/cmake-modules
+	${CMAKE_MODULE_PATH}
+)
+
+if(NOT CMAKE_BUILD_TYPE)
```

Comment:
if you're referring directly at the if statement, perhaps not, if anything maybe a more appropriate check would be `AND NOT CMAKE_CONFIGURATION_TYPES`.
however, `CMAKE_BUILD_TYPE` would still need to be set to choose the correct `CMAKE_CXX_FLAGS_*` for single config build (as per last answer). keeping in mind that if the `if` and `set` is removed and the user doesn't specify build type, it effectively defaults to `?debug?` since `assert` won't be disabled.